woocommerce/woocommerce
# Allow running of "Smoke test release" on draft releases

### All Submissions:


### Changes proposed in this Pull Request:

Closes #36998.

This PR would allow the `Smoke test release` workflow to be run on a draft release. Also, it contains a few several fixes in the workflow file.

The main reason why the `Smoke test release` workflow previously couldn't run on draft releases was because it's using the default `github.token` when retrieving the list of releases. Since this token has limited permissions, it can only access releases/pre-releases that are already published.

To fix this, the `Smoke test release` workflow now uses the same `E2E_GH_TOKEN` used in other workflows. With this token, draft releases can already be accessed.

### How to test the changes in this Pull Request:


1. Create a draft release. Make sure it has:
   1. A `woocommerce.zip` asset, and
   1. A temporary tag for testing purposes, like `test-pr-36997`.
1. Go to the [`Smoke test release`](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-release.yml) page.
1. In the "Use workflow from" dropdown, select this branch `e2e/release-include-drafts`.
1. In the "WooCommerce Release Tag" box, enter the tag you created in step 1.
1. Click "Run workflow" to manually start it.
1. Wait for it to finish. All jobs should be executed like so:
       <img width="202" alt="yAS4WTxv76" src="https://user-images.githubusercontent.com/4509348/222337758-3c228f5a-7ba4-4521-aa76-ca1807fafe76.png">
1. Go to [WooCommerce Test Reports > Smoke Tests on Releases](https://woocommerce.github.io/woocommerce-test-reports/release/) page.
1. You should see the tag listed in the page, and that the API and E2E test results are displayed, regardless of whether they passed or failed. 
   <img width="413" alt="Mp0drcndSH" src="https://user-images.githubusercontent.com/4509348/222337904-d1ee9d9c-f057-49b8-ad0c-9a21afd49f6f.png">
   If you're taken to a 404 page, chances are the reports are still being uploaded. Wait for a few more minutes, then refresh the page.
1. Delete the draft release after testing.

### Other information:


### FOR PR REVIEWER ONLY:
